sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -555,12 +557,12 @@ object GpuToTimestamp extends Arm {
     // we support date formats using either `-` or `/` to separate year, month, and day and the
     // regex rules are written with '-' so we need to replace '-' with '/' here as necessary
     val rulesWithSeparator = format.separator match {
-      case '/' =>
+      case Some('/') =>
         regexReplaceRules.map {
           case RegexReplace(pattern, backref) =>
             RegexReplace(pattern.replace('-', '/'), backref.replace('-', '/'))
         }
-      case '-' =>
+      case Some('-') | Some(_) | None =>
```

Comment:
It looks like this is an exhaustive check for all other cases so could be simplified:

```suggestion
      case _ =>
```